src/sql-server-util/src/cdc.rs

```diff
@@ -175,7 +201,7 @@ impl<'a> CdcStream<'a> {
                             self.client,
                             &*instance,
                             *instance_lsn,
-                            new_lsn,
+                            db_max_lsn,
                             RowFilterOption::AllUpdateOld,
                         )
                         // TODO(sql_server3): Make this chunk size configurable.
```

Comment:
Not part of this PR, but the code below seems to be doing some grouping by LSNs. Timely doesn't care about the order of LSNs in the data stream as long as progress statements are correct so we can directly iterate and emit the data as they come.

That said, in which case does SQL Server send us data out of order?

Reply:
Good to know!

SQL Server shouldn't be sending data out of order but the type this Stream yields is:
```
Data {
    /// The capture instance these changes are for.
    capture_instance: Arc<str>,
    /// The LSN that this change occurred at.
    lsn: Lsn,
    /// The change itself.
    changes: Vec<Operation>,
},
```
So we're grouping by LSN just so we can emit an entire set of `Operation`s all at once instead of `(Arc<str>, Lsn, Operation)` tuples. Happy to refactor this in a follow up if you'd prefer!

Reply:
Emitting a sequence of `(Arc<str>, Lsn, Operation)` is definitely more timely-like so I highly suggest that we do that, and totally fine as a followup. Separately, it would be nice if we could get rid of that `Arc<str>` in favor of the table id which I presume is some small copy-able type that doesn't need refcounts update all the time.